### PR TITLE
Add preonic/kjwon15 layout

### PR DESCRIPTION
**Pull request by Kjwon15**
## Description

It Provides these
- Mouse layer (Press left space to use it)
- Almost full functionality for Linux
- Dual function keys are not holdable, so I placed it on another place with single function
  - For example, Enter is placed with right shift, So If you want to Press and hold Enter key, Use left bottom button
- Blank layout for some creative usage: like cleaning your keyboard without disconnect USB
- Pure mode that doesn't use any dual functioning keys for gaming

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fix music mode keymap (See `music_map` on `preonic.c`)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/preonic/keymaps/kjwon15/keymap.c [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+      } else {
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      register_code(KC_PGDN);
+      unregister_code(KC_PGDN);
+    } else {
+      register_code(KC_PGUP);
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/kjwon15/keymap.c]:
```suggestion
      tap_code(KC_PGUP);
```

File: keyboards/preonic/keymaps/kjwon15/keymap.c [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+      unregister_code(KC_PGUP);
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/kjwon15/keymap.c]:
```suggestion
```

File: keyboards/preonic/keymaps/kjwon15/keymap.c [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+          if (record->event.pressed) {
+            layer_on(_RAISE);
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          } else {
+            layer_off(_RAISE);
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          }
+          return false;
+          break;
+        case BACKLIT:
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/kjwon15/keymap.c]: You're not using this keycode, so this whole `case` block can be safely removed.

File: keyboards/preonic/keymaps/kjwon15/keymap.c [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+  _MOUSE
+};
+
+enum preonic_keycodes {
+  QWERTY = SAFE_RANGE,
+  PURE,
+  BLANK,
+  LOWER,
+  RAISE,
+  BACKLIT,
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/kjwon15/keymap.c]:
```suggestion
```

File: keyboards/preonic/keymaps/kjwon15/config.h [hunk 5]
```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "config_common.h"
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/kjwon15/config.h]:
```suggestion
```

**Comment by Kjwon15** [comment]
Anything else that I should change?

File: keyboards/preonic/keymaps/kjwon15/keymap.c [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+      } else {
+        layer_off(_ADJUST);
+      }
+      break;
+    case 1:
+      if (active) {
+        muse_mode = true;
+      } else {
+        muse_mode = false;
+        #ifdef AUDIO_ENABLE
```
Comment (drashna) [inline keyboards/preonic/keymaps/kjwon15/keymap.c]: There are some changes to the default keymap that fix some issues with dip switch, and the startup sound.   You should take a look at those changes (to both this part, and to the matrix scan/muse stuff).

Comment (Kjwon15) [reply keyboards/preonic/keymaps/kjwon15/keymap.c]: Thank you!

**Pull request merged**